core/src/main/java/com/datastrato/gravitino/storage/relational/service/GroupMetaService.java

```diff
@@ -0,0 +1,236 @@
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+/** The service class for group metadata. It provides the basic database operations for group. */
+public class GroupMetaService {
+  private static final GroupMetaService INSTANCE = new GroupMetaService();
+
+  public static GroupMetaService getInstance() {
+    return INSTANCE;
+  }
+
+  private GroupMetaService() {}
+
+  private GroupPO getGroupPOByMetalakeIdAndName(Long metalakeId, String groupName) {
+    GroupPO GroupPO =
+        SessionUtils.getWithoutCommit(
+            GroupMetaMapper.class,
+            mapper -> mapper.selectGroupMetaByMetalakeIdAndName(metalakeId, groupName));
+
+    if (GroupPO == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE,
+          Entity.EntityType.GROUP.name().toLowerCase(),
+          groupName);
+    }
+    return GroupPO;
+  }
+
+  private Long getGroupIdByMetalakeIdAndName(Long metalakeId, String groupName) {
+    Long groupId =
+        SessionUtils.getWithoutCommit(
+            GroupMetaMapper.class,
+            mapper -> mapper.selectGroupIdBySchemaIdAndName(metalakeId, groupName));
+
+    if (groupId == null) {
+      throw new NoSuchEntityException(
+          NoSuchEntityException.NO_SUCH_ENTITY_MESSAGE,
+          Entity.EntityType.GROUP.name().toLowerCase(),
+          groupName);
+    }
+    return groupId;
+  }
+
+  public GroupEntity getGroupByIdentifier(NameIdentifier identifier) {
+    Preconditions.checkArgument(
+        identifier != null
+            && !identifier.namespace().isEmpty()
+            && identifier.namespace().levels().length == 3,
+        "The identifier should not be null and should have three level.");
+    Long metalakeId =
+        MetalakeMetaService.getInstance().getMetalakeIdByName(identifier.namespace().level(0));
+    GroupPO groupPO = getGroupPOByMetalakeIdAndName(metalakeId, identifier.name());
+    List<RolePO> rolePOs = RoleMetaService.getInstance().listRolesByGroupId(groupPO.getGroupId());
+
+    return POConverters.fromGroupPO(groupPO, rolePOs, identifier.namespace());
+  }
+
+  public void insertGroup(GroupEntity GroupEntity, boolean overwritten) {
+    try {
+      Preconditions.checkArgument(
+          GroupEntity.namespace() != null
+              && !GroupEntity.namespace().isEmpty()
+              && GroupEntity.namespace().levels().length == 3,
+          "The identifier should not be null and should have three level.");
+
+      Long metalakeId =
+          MetalakeMetaService.getInstance().getMetalakeIdByName(GroupEntity.namespace().level(0));
+      GroupPO.Builder builder = GroupPO.builder().withMetalakeId(metalakeId);
+      GroupPO GroupPO = POConverters.initializeGroupPOWithVersion(GroupEntity, builder);
+
+      List<Long> roleIds = GroupEntity.roleIds();
+      if (roleIds == null) {
```

Comment:
As `roleNames` and `rolesIds` are identical in null or not null,  so I wander if we need the logic here.

Reply:
ok, i will optimize it